goal/src/main/java/org/openapitools/vertxweb/server/api/PetsApi.java

```diff
@@ -26,6 +29,12 @@ static PetsApi create(RouterBuilder builder) {
     @Fluent
     PetsApi listPetsHandler(Handler<RoutingContext> handler);
 
+    /**
+     * Generate generic javadoc, stating that the API params are parsed into the expected type in the context data
+     */
+    @Fluent
+    <T> PetsApi listPets(Function<ListPetsArguments, Future<T>> function);
```

Comment:
Accept a functional interface codegenerated more than Function?

Reply:
And also, in addition to this signature, what about accepting a function that doesn't return anything but provides routing context and the extracted params?

Reply:
I think you're going in the same direction of my comments above! I agree, having a forced return has some drawbacks and limits the composition.